tests/unit/test_charm.py

```diff
@@ -1735,479 +1728,470 @@ def test_juju_run_exec_divergence(self, _juju_version: Mock, _topology_observer:
-        self.harness.charm.set_secret(scope, "new-secret", "blablabla")
-        assert self.harness.charm.get_secret(scope, "new-secret") == "blablabla"
+        harness.charm.set_secret(scope, "new-secret", "blablabla")
+        assert harness.charm.get_secret(scope, "new-secret") == "blablabla"
 
         # Set another new secret
-        self.harness.charm.set_secret(scope, "new-secret2", "blablabla")
-        assert self.harness.charm.get_secret(scope, "new-secret2") == "blablabla"
-
-    @parameterized.expand([("app", True), ("unit", True), ("unit", False)])
-    @patch_network_get(private_address="1.1.1.1")
-    @patch("charm.PostgresqlOperatorCharm._on_leader_elected")
-    @patch("charm.JujuVersion.has_secrets", new_callable=PropertyMock, return_value=True)
-    def test_invalid_secret(self, scope, is_leader, _, __):
+        harness.charm.set_secret(scope, "new-secret2", "blablabla")
+        assert harness.charm.get_secret(scope, "new-secret2") == "blablabla"
+
+
+@pytest.mark.parametrize("scope,is_leader", [("app", True), ("unit", True), ("unit", False)])
+@patch_network_get(private_address="1.1.1.1")
+def test_invalid_secret(harness, scope, is_leader):
+    with (
+        patch("charm.PostgresqlOperatorCharm._on_leader_elected"),
+    ):
+        harness.add_relation(PEER, harness.charm.app.name)
         # App has to be leader, unit can be either
-        self.harness.set_leader(is_leader)
+        harness.set_leader(is_leader)
 
-        with self.assertRaises(RelationDataTypeError):
-            self.harness.charm.set_secret(scope, "somekey", 1)
+        with pytest.raises(RelationDataTypeError):
+            harness.charm.set_secret(scope, "somekey", 1)
 
-        self.harness.charm.set_secret(scope, "somekey", "")
-        assert self.harness.charm.get_secret(scope, "somekey") is None
+        harness.charm.set_secret(scope, "somekey", "")
+        assert harness.charm.get_secret(scope, "somekey") is None
 
-    @pytest.mark.usefixtures("use_caplog")
-    @patch_network_get(private_address="1.1.1.1")
-    @patch("charm.PostgresqlOperatorCharm._on_leader_elected")
-    def test_delete_password(self, _):
-        """NOTE: currently ops.testing seems to allow for non-leader to remove secrets too!"""
-        self.harness.set_leader(True)
-        self.harness.update_relation_data(
-            self.rel_id, self.charm.app.name, {"replication": "somepw"}
-        )
-        self.harness.charm.remove_secret("app", "replication")
-        assert self.harness.charm.get_secret("app", "replication") is None
 
-        self.harness.set_leader(False)
-        self.harness.update_relation_data(
-            self.rel_id, self.charm.unit.name, {"somekey": "somevalue"}
-        )
-        self.harness.charm.remove_secret("unit", "somekey")
-        assert self.harness.charm.get_secret("unit", "somekey") is None
+@patch_network_get(private_address="1.1.1.1")
+def test_delete_existing_password_secrets(harness, _has_secrets, caplog):
```

Comment:
IMO, we may rename this test `test_delete_password` as it's being executed for both secret-enabled and secrets-not-enabled Juju versions. I'd also do the same for `test_update_certificate_secrets` (which I'd rename `test_update_certificate`).